.github/workflows/test-pypi-publish.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    tags:
+      - 'v*rc*'
+
+jobs:
+  build-n-publish:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install build dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+
+      - name: Install dependencies
+        run: |
+          python -m pip install .[dev]
+
+      - name: Check manifest
+        run: |
+          invoke check-manifest
+
+      - name: Build package
+        run: |
+          python -m build .
+
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository-url: https://test.pypi.org/legacy/
```

Comment:
test.pypi.org is neat 📸 Any reason you're using the legacy endpoint here?

Reply:
It's the recommended `repository-url` of the gh-action-pypi-publish: https://github.com/pypa/gh-action-pypi-publish , so I followed their docs.